project/driver_controls/inc/input_event.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+// Define common typedefs for the various FSMs to use.
+
+#include "event_queue.h"
+
+// Typedef for the FSM arbitration functions
+typedef bool (*InputEventCheck)(const Event *e);
```

Comment:
I'm really not a fan of having this here. It really belongs in `event_arbiter.h`.
Just include the header in your FSM source files so you don't need to expose it to the user.